integreat_cms/cms/utils/task_queue.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from queue import Empty, LifoQueue
+import threading
+
+logger = logging.getLogger(__name__)
+
+class TaskQueue:
```

Comment:
Hmm, I'm not sure whether this use case really justifies a queue – after all, this link replacement function is only executed once per region cloning process, right? So I think it will be an extreme edge case that two regions are cloned at the same time and even if this happens, putting those two jobs into a queue wouldn't make much of a difference, I mean probably the normal CPU scheduler is capable of handling two background threads, right? :thinking:

Maybe I'm missing something, but I think simply this should do it:
```
    t = threading.Thread(
        target=replace_links,
        kwargs={
            "search": old_link,
            "replace": new_link,
            "region": region,
            "link_types": ["internal"],
        },
        daemon=True,
        name="replace_links",
    )
    t.start()
```

Or, if this does not work for some reason, I think we don't need to re-define the queue in order to use it – we could just import the linkcheck queue and add tasks to it.

Reply:
Jep that's actually what I started with.
However, I realized, that we need to have a way to check whether the thread has been finished in the tests. 
I figured that there are 2 ways to do this:
1. Add a check `if TESTING: t.join()` to run the thread synchronously in the tests.
2. Add a helper class

I chose the second option as I thought it could be helpful in the future to have a manager for background tasks in general. If that's not relevant rn I'm open to remove this.

Reply:
Ah ok, didn't realize this.
Hmm, I'm undecided on this, but leaning a bit towards the first solution. Maybe just a new setting `ENABLE_BACKGROUND_TASKS` or similar which is `True` by default but can be disabled during tests or when debugging problems on the production server?
I'm not yet sure whether we can reuse the queue – it depends a bit on our future use cases, I guess if we implement more complex background tasks, we would be better off relying on a queue worker that is independent from the wsgi processes that handle the webserver requests, e.g. by using Celery or similar. Also, we did not yet find a process-safe locking mechanism which could also influence this decision. So as long as we don't know in which direction we're heading, I would vote for removing the queue, but I'm also fine with keeping it for the moment.

Reply:
Sure. Fine with me. Especially as there is no actual use rn.